src/test/kotlin/uk/gov/dluhc/printapi/database/repository/PrintDetailsRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,162 @@
+package uk.gov.dluhc.printapi.database.repository
+
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import uk.gov.dluhc.printapi.database.entity.Status
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidRequestId
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintDetails
+import java.util.UUID
+
+internal class PrintDetailsRepositoryIntegrationTest : IntegrationTest() {
+
+    @Nested
+    inner class Save {
+        @Test
+        fun `should save and get print details by id`() {
+            // Given
+            val details = buildPrintDetails()
+
+            // When
+            printDetailsRepository.save(details)
+
+            // Then
+            val saved = printDetailsRepository.get(details.id!!)
+            assertThat(saved).usingRecursiveComparison().isEqualTo(details)
+        }
+    }
+
+    @Nested
+    inner class Get {
+        @Test
+        fun `should throw exception given non-existing print details`() {
+            // Given
+            val id = UUID.randomUUID()
+
+            // When
+            val ex = Assertions.catchThrowableOfType(
+                { printDetailsRepository.get(id) },
+                PrintDetailsNotFoundException::class.java
+            )
+
+            // Then
+            assertThat(ex).isNotNull.hasMessage("Print details not found for id: $id")
+        }
+    }
+
+    @Nested
+    inner class GetByRequestId {
+        @Test
+        fun `should return print details given existing print details with requestId`() {
+            // Given
+            val expected = buildPrintDetails()
```

Comment:
Minor - is it worth specifying the `requestId` on each of these builder calls, just so that its clear whats different about each one and why (because it is the relevant factor to this test)

Reply:
Sure, @NathanRussellValtech ,
I will address it.